### ISSUE TITLE
New initramfs hook that uses LEDs etc. to let us know pmOS booted (when networking and screen don't work)

### ISSUE THREAD

**Issue by ollieparanoid** [postmarketOS/pmbootstrap#1217]
This idea was brought up the other day in #postmarketOS: An initramfs hook that tries to notify the user with alternative means that the kernel is actually booted:
* let the LEDs blink
* use the vibration motor
* use the flashlight
* ...

When the device does not respond over USB, and the screen does not work either, this would be an easy way to verify that postmarketOS has booted on the device.

See also: [Troubleshooting:boot - Does the kernel boot?](https://wiki.postmarketos.org/wiki/Troubleshooting:boot#Does_the_kernel_boot.3F)

**Comment by MayeulC** [comment]
I can give it a go, I would need it right about now.

**Comment by eMPee584** [comment]
Just for inspiration, I have something similar on my debianized qwerty android phone.. from my `/system/etc/init.d/99debian` (a lil' cleaned):

```
# blink the sh*t until boot complete
echo 1 > /sys/class/misc/backlightnotification/enabled
{ 
  while [ ! "$(getprop sys.boot_completed)" == "1" ] || [ ! "$(getprop init.svc.bootanim)" == "stopped" ] ; do
    echo 1 > /sys/class/misc/backlightnotification/notification_led
    sleep 0.05
    echo 0 > /sys/class/misc/backlightnotification/notification_led
    sleep 1
  done
  vibrate 5 200 0.6
  echo 0 > /sys/class/misc/backlightnotification/enabled
} &

```

**Comment by MayeulC** [comment]
Thanks @eMPee584 for your input. It looks like the values you have here would be incompatible with the script I wrote. Are you running a mainline kernel? I thought every LED would be accessed trough the standard LED API (although I haven't been able to turn on the notification led again on my phone). This one includes a `brightness` file descriptor, as well as a `max_brightness` one, as far as I know.

The vibrate command is interesting as well... To be fair, I would be interested in peering at your sysfs tree :blush:

**Comment by eMPee584** [comment]
yeah, the actual LED access is definitely all non-standard and OEM dependent.. what I deemed useful to show is the outer backgrounding loop.. but don't let it get in the way of spinning it your own preferred style :+1: 
my `/system/xbin/vibrate` very simply contains

```
#!/system/bin/sh
for duration in $@; do
  echo ${duration} > /sys/devices/virtual/timed_output/vibrator/enable
  sleep 0.1
done
```

**Comment by eMPee584** [comment]
oh & by the way: actually, the `backlightnotification/notification_led` is the backlight of the display button row (back, home, menu, search) on this model (i927) — it doesn't have a proper notification LED.

**Comment by MayeulC** [comment]
`/sys/devices/virtual/timed_output/vibrator/enable`
That's nice to know, my device has the same interface (plus one at `/sys/class/timed_output/vibrator/`, that I used in the implementation, see https://github.com/postmarketOS/pmbootstrap/pull/1238/files ).
